openlibrary/core/bookshelves.py

```diff
@@ -404,25 +509,35 @@ def get_sorted_reading_log_books(
                 oldb.query(query, vars=query_params)
             )
 
-            reading_log_work_keys = [
-                '/works/OL%sW' % i['work_id'] for i in reading_log_books
+            reading_log_keys = [
+                ('/works/OL%sW' % i['work_id'], '/books/OL%sM' % i['edition_id'])
+                if editions and i['edition_id']
+                else ('/works/OL%sW' % i['work_id'], None)
+                for i in reading_log_books
             ]
+
             solr_docs = get_solr().get_many(
-                reading_log_work_keys,
+                [key for key_pair in reading_log_keys for key in key_pair if key],
```

Comment:
We might be able to use the `flatten` method in our vendor/infogami/infogami/infobase/utils.py

```suggestion
                flatten(reading_log_keys),
```

Reply:
Flatten has been used. The infogami implementation of it limits it to lists, so I was forced to change the tuple here to a list. Fortunately, that causes it to work just fine.